### PR TITLE
Update to pyiron_base>=0.1.40

### PR DESCRIPTION
**Pull request by pmrv**

### CODE REVIEW

**Comment by jan-janssen** [comment]
The dependency in `setup.py` are ignored during the tests - for the tests we have to update https://github.com/pyiron/pyiron/blob/master/.ci_support/environment.yml 

Still knowing this it might make sense to add a `pip check` to the tests.

**Comment by jan-janssen** [comment]
I created the corresponding pull request here https://github.com/pyiron/pyiron/pull/1270 - it is merged with `pmrv-patch-1` rather than the `master` branch.

**Comment by pmrv** [comment]
I guess the package is not distributed by conda yet.  It already shows up in the file [list](https://anaconda.org/conda-forge/pyiron_base/files), so I'll just wait a bit.

**Comment by jan-janssen** [comment]
I just triggered the builds again.

File: tests/testing/test_exampleJob.py [hunk 1]
```diff
@@ -30,9 +30,7 @@ def tearDownClass(cls):
 
     def test_delete_existing_job(self):
         self.job = self.project.create_job("ExampleJob", "job_test_delete_existing_job")
-        with warnings.catch_warnings(record=True) as w:
-            self.job.run(run_again=True)
-            self.assertEqual(len(w), 1)
```
Comment (jan-janssen) [inline tests/testing/test_exampleJob.py:35]: It is strange that the warning message disappeared.

Comment (pmrv) [reply tests/testing/test_exampleJob.py:35]: I assume  dac53480d2f9b200e0c4b77e60df97a17f7831b3 is the culprit, will have a look.

Comment (pmrv) [reply tests/testing/test_exampleJob.py:35]: It's a deeper bug in the deprecator, I'll close this and bump the pyiron_base version again after I fix it there.

**Pull request closed without merging**